components/identity-core/org.wso2.carbon.identity.core.ui/src/main/resources/web/identity/validation/js/identity-validate.js

```diff
@@ -0,0 +1,218 @@
+ * This file expects dialog/js/dialog.js to be included while including this script.
+ */
+
+/**
+ * Returns the regular expression for the provided pattern key.
+ * If no regex is defined for the provided input returns the provided input as a regex
+ *
+ * @param pattern regex pattern key or undefined regex as a string
+ */
+function getPattern(pattern) {
+    var regex;
+    switch (pattern) {
+        case "digits-only":
+            regex = /^[0-9]+/;
+            break;
+        case "alphabetic-only":
+            regex = /^[a-zA-Z]+/;
+            break;
+        case "alphanumerics-only":
+            regex = /^[a-zA-Z0-9]+/;
+            break;
+        case "url":
+            regex = /^(([^:/?#]+):)?(([^/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?/;
+            break;
+        case "email":
+            regex = /^(.+)@(.+)$/;
+            break;
+        case "whitespace-exists":
+            regex = /.*\s+.*/;
+            break;
+        case "uri-reserved-exists":
+            regex = /.*[:/\?#\[\]@!\$&'\(\)\*\+,;=]+.*/;
+            break;
+        case "uri-unsafe-exists":
+            regex = /.*[<>%\{\}\|\^~\[\]`]+.*/;
+            break;
+        case "html-meta-exists":
+            regex = /.*[&<>"'/]+.*/;
+            break;
+        case "xml-meta-exists":
+            regex = /.*[&<>"']+.*/;
+            break;
+        default:
+            regex = new RegExp(pattern);
+            break;
+    }
+
+    return regex;
+}
+
+function isWhiteListed(input, whiteListPatterns) {
+    var isValid = false;
+    var pattern;
+    for (var i = 0; i < whiteListPatterns.length; i++) {
+        pattern = getPattern(whiteListPatterns[i]);
+        isValid = pattern.test(input);
+        if (isValid) {
+            break;
+        }
+    }
+
+    return isValid;
+}
+
+function isNotBlackListed(input, blackListPatterns) {
+    var isValid = false;
+    var pattern;
+    for (var i = 0; i < blackListPatterns.length; i++) {
+        pattern = getPattern(blackListPatterns[i]);
+        isValid = !pattern.test(input);
+
+        if (!isValid) {
+            break;
+        }
+    }
+
+    return isValid;
+}
+
+function getPatternString(patterns) {
+    var patternString = "";
+    for (var i = 0; i < patterns.length; i++) {
+        patternString += getPattern(patterns[i]).toString();
+        if ((patterns.length - 1) != i) {
+            patternString += ", ";
+        }
+    }
+
+    return patternString;
+}
+
+/**
+ * Validates all input elements of the form which defines attributes for white list or black list patterns against them.
+ *
+ * @param form Form to be validated
+ * @returns {{isValid: boolean, label: string, whiteListPatterns: string, blackListPatterns: string}}
+ */
+function validate(form) {
+    var allInputs = form.getElementsByTagName('input');
+    var len = allInputs.length;
+
+    var whiteListPatternString = "";
+    var blackListPatternString = "";
+    var labelString = "";
+
+    for (var i = 0; i < len; i++) {
```

Comment:
can we move the processing of input element to new method so that anyone call that if needed without giving the whole form ?